client/src/state/snapshot.rs

```diff
@@ -130,13 +130,13 @@ impl Snapshot {
     /// Deserialize, encrypt and compress the new state to a bytestring.
     pub fn export_to_serialized_state(
         &mut self,
-        diff: <sync::SnapshotState as MergeLayer>::Hierarchy,
+        diff: <SyncSnapshotState as MergeLayer>::Hierarchy,
```

Comment:
Better pass it as generic: `<SyncSnapshotState as MergeLayer>::Hierarchy` could some function generic `H`

Reply:
hm not quite sure to use an inner type as an outer type :)

Reply:
Good point, I agree that it is weird to expose to associated type of an inner trait like this. I changed it in 4af3773 to just use the actual types directly.